r/vignettes/python.Rmd

```diff
@@ -0,0 +1,127 @@
+---
+title: "Arrow in Python and R with reticulate"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Using Arrow in Python and R with reticulate}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+The `arrow` package provides `reticulate` methods for passing data between
+R and Python. This document provides a brief overview.
+
+## Installing
+
+To use `arrow` in Python, at a minimum you'll need the `pyarrow` library.
+To install it in a virtualenv,
+
+```r
+library(reticulate)
+virtualenv_create("arrow")
+virtualenv_install("arrow", "pyarrow")
+```
+
+If you want to install a development version of `pyarrow`, you can shell out
+and call `pip` directly.
+
+```r
+system(
+  paste(
+    file.path(virtualenv_root(), "arrow/bin/python"), "-m pip install --extra-index-url https://repo.fury.io/arrow-nightlies/ --pre pyarrow --force-reinstall"
+  )
+)
+```
+
+For more on installing and configuring Python, including using `conda`,
+see the [reticulate docs](https://rstudio.github.io/reticulate/articles/python_packages.html).
+
+## Using
+
+To start, load `arrow` and `reticulate`, and then import `pyarrow`.
+
+```r
+library(arrow)
+library(reticulate)
+use_virtualenv("arrow")
+pa <- import("pyarrow")
+```
+
+The package includes support for transferring Arrow `Array` and `RecordBatch`
+between R and Python. For example, let's create an `Array` in `pyarrow`.
+
+```r
+a <- pa$array(c(1, 2, 3))
+a
+
+## Array
+## <double>
+## [
+##   1,
+##   2,
+##   3
+## ]
+```
+
+`a` is now an `Array` object in our R session, even though we created it in Python.
+We can apply R methods on it:
+
+```r
+a[a > 1]
+
+## Array
+## <double>
+## [
+##   2,
+##   3
+## ]
+```
+
+We can send data both ways. One reason we might want to use `pyarrow` in R is
+to take advantage of functionality that is better supported in Python than in R.
+For example, `pyarrow` has a `concat_arrays` function, but as of 0.17, this
+function is not implemented in the `arrow` R package. We can use `reticulate`
+to use it efficiently.
+
+```r
+b <- Array$create(c(5, 6, 7, 8, 9))
+a_and_b <- pa$concat_arrays(r_to_py(list(a, b)))
+a_and_b
+
+## Array
+## <double>
+## [
+##   1,
+##   2,
+##   3,
+##   5,
+##   6,
+##   7,
+##   8,
+##   9
+## ]
+```
+
+Now we have a single `Array` in R.
+
+"Send", however, isn't the correct word. Internally, we're passing pointers to the
+data, without copying anything. Nothing is being sent: we're transferring
```

Comment:
Maybe to be explicit say "passing pointers to the data between the R and Python interpreters running together in the same process"